src/components/utils.tsx

```diff
@@ -0,0 +1,45 @@
+interface ElementObject {
+  id: string;
+}
+export enum ElementType {
+  RACK = "racks",
+  INSTANCE = "instances",
+  MODEL = "models"
+}
+export interface InstanceObject extends ElementObject {
+  hostname: string;
+  elevation: number;
+  model: ModelObject;
+  rack: RackObject;
+  owner: string;
+  comment: string;
+}
+
+export interface RackObject extends ElementObject {
+  row_letter: string;
+  rack_num: number;
+  height: number;
+}
+
+export interface ModelObject extends ElementObject {
+  vendor: string;
+  model_number: string;
+  height: number;
+  display_color: string | undefined;
+  num_ethernet_ports: number | undefined;
+  num_power_ports: number | undefined;
+  cpu: string | undefined;
+  memory_gb: number | undefined;
+  storage: string | undefined;
+  comment: string | undefined;
+}
+export type ElementObjectType = ModelObject | RackObject | InstanceObject;
+export function isModelObject(obj: any): obj is ModelObject {
+  return obj && obj.model_number;
+}
+export function isRackObject(obj: any): obj is RackObject {
+  return obj && obj.rack_num;
+}
+export function isInstanceObject(obj: any): obj is InstanceObject {
+  return obj && obj.hostname;
+}
```

Comment:
This is the most important part of the PR to look at !! Typescript does not have the traditional objects found in java, but it DOES support objects that extend certain interfaces. This is the most useful for mapping axios response objects into objects that extend these interfaces. Also added three functions that can be helpful to check the type of object that you are working with.

Reply:
Can you explain how the functions work? I get how the type predicating works with `obj is ________` but what's going on with the return values?

Reply:
The return properties are kind of arbitrary, but the idea is that im making the assumptions model_number is a unique property ModelObject, so that any object with that property is a ModelObject, so if the property exists, the function will return true. This could be modified with more properties if necessary though